src/containers/Header/components/DateButton/index.js

```diff
@@ -1,25 +1,31 @@
 // @flow
 
-import React from 'react';
+import React, { useCallback } from 'react';
 import type { Dispatch } from 'redux';
-import type { dateAction } from '../../../../types/redux';
+import type { DateAction } from '../../../../types/redux';
 
 type Props = {
-  onClick: () => Dispatch<dateAction>,
+  onClick: () => Dispatch<DateAction>,
   label: string,
   active: boolean
 };
 
-const DateButton = (props: Props) => (
-  <button
-    type="button"
-    disabled={!props.active}
-    onClick={() => {
-      props.onClick();
-    }}
-  >
-    {props.label}
-  </button>
-);
+const DateButton = (props: Props) => {
+  const { onClick } = props;
```

Comment:
In case of functional components you can decompose props right on the start, so you don't need to use `props` variable

```suggestion
const DateButton = ({ onClick, label, active }: Props) => {
  const { onClick } = props;
```